app/code/core/Mage/Sales/sql/sales_setup/upgrade-1.6.0.10-1.6.0.11.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * OpenMage
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@magento.com so we can send you a copy immediately.
+ *
+ * @category    Mage
+ * @package     Mage_Sales
+ * @copyright  Copyright (c) 2006-2020 Magento, Inc. (http://www.magento.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/* @var Mage_Sales_Model_Entity_Setup $installer */
+$installer = $this;
+
+$installer->startSetup();
+
+// Add index to sales_flat_order on customer_email for fast lookup, only first 15 bytes
+$keyList = $installer->getConnection()->getIndexList($installer->getTable('sales/order'));
+if (!isset($keyList[strtoupper('customer_email')])) {
+    $installer->run("
+        ALTER TABLE {$installer->getTable('sales/order')}
+        ADD INDEX `IDX_SALES_FLAT_ORDER_CUSTOMER_EMAIL` (`customer_email` (15));
+    ");
+}
```

Comment:
```suggestion
if (!isset($keyList['IDX_SALES_FLAT_ORDER_CUSTOMER_EMAIL'])) {
    $installer->run("
        ALTER TABLE {$installer->getTable('sales/order')}
        ADD INDEX `IDX_SALES_FLAT_ORDER_CUSTOMER_EMAIL` (`customer_email` (15));
    ");
}
```

What is the correct syntax for using ```(`customer_email` (15))``` with `$this->getConnection()->addIndex()`)